scripts/exact-example.jl

```diff
@@ -0,0 +1,245 @@
+
+# KrylovKit's
+# [`eigsolve`](https://jutho.github.io/KrylovKit.jl/stable/man/eig/#KrylovKit.eigsolve)
+# function is implemented in a way that does not require the linear operator and vector to
+# be Julia arrays. Rimu leverages this functionality, which allows diagonalising
+# Hamiltonians without ever needing to construct the matrix - all matrix elements are
+# generated on the fly.
+
+# While this method is by far the slowest of the ones discussed, it also uses drastically
+# less memory. This allows us to diagonalise much larger Hamiltonians.
+
+# To use this method, you first need a starting [`DVec`](@ref):
+
+dvec = DVec(add => 1.0)
+
+# Then, pass that vector and the Hamiltonian to
+# [`eigsolve`](https://jutho.github.io/KrylovKit.jl/stable/man/eig/#KrylovKit.eigsolve). Since
+# the function has no way of knowing the Hamiltonian is Hermitian, we have to provide that
+# information through the `issymmetric` or `ishermitian` keyword arguments. Make sure to
+# only pass this argument when the Hamiltonian is actually symmetric. To check that,
+# look at the `LOStructure` trait:
+
+LOStructure(ham)
+
+#
+
+vals_mf, vecs_mf = eigsolve(ham, dvec, num_eigvals, :SR; issymmetric=true)
+vals_mf
+
+# Keep in mind that if an eigenvector is orthogonal to `dvec`, KrylovKit will miss
+# it. Consider the following example:
+
+eigsolve(ham, vecs_mf[2], num_eigvals, :SR, issymmetric=true)[1]
+
+# ## Reducing matrix size with symmetries
+
+# As these matrices tend to get large quickly, memory is usually the bottleneck.  There are
+# currently two methods implemented to reduce the matrix size, [`ParitySymmetry`](@ref) and
+# [`TimeReversalSymmetry`](@ref). These symmetries work by performing a similarity transform
+# on the Hamiltonian which causes it to become block-diagonal. When building a matrix from a
+# block-diagonal Hamiltonian, only the block that contains the starting address is
+# constructed.
+
+# You should only use these where the relevant symmetries actually apply - no checks are
+# performed to make sure they do. There is also currently no way of using both at the same
+# time. Please consult the documentation for a more in-depth description of these options.
+
+# The Hamiltonian presented in this example is compatible with the
+# [`ParitySymmetry`](@ref). Let's see how the matrix size is reduced when applying it.
+
+size(sparse(ham))
+
+#
+
+size(sparse(ParitySymmetry(ham)))
+
+# In this small example, the size reduction is modest, but for larger systems, you can
+# expect to reduce the dimension of the matrix by about half.
+
+all_eigs = eigvals(Matrix(ham))
+even_eigs = eigvals(Matrix(ParitySymmetry(ham)))
+
+# The eigenvalues of the transformed Hamiltonian are a subset of the full spectrum. To get
+# the other half, we can pass the `even=false` keyword argument to it. When doing that, we
+# need to make sure the starting address of the Hamiltonian is odd. In this case, an odd
+# address is one that is not symmetric under reversal:
```

Comment:
I don't think I would call it odd unless it is odd - which cannot happen for an onr.
```suggestion
# need to make sure the starting address of the Hamiltonian is not symmetric under reversal:
```